feat(frontend):Updated pages with relatedItems menu

nkrmr opened the pull request:
- Added related items sticky menu on pages
- Created getSearch method to request search into API
- Used getSearch into Items request to get relatedItems
- Removed narrow style to feedback block
- Updated snapshot

nkrmr committed:
wip

nkrmr committed:
wip

nkrmr committed:
wip

nkrmr committed:
wip

nkrmr committed:
wip

nkrmr committed:
feat(frontend):Updated pages with relatedItems menu

nkrmr committed:
feat(frontend):Updated pages with relatedItems menu

nkrmr committed:
feat(frontend):Updated pages with relatedItems menu

nkrmr committed:
feat(api):Created getSearch method to make search request and added thhis one within items request to get related items

nkrmr committed:
feat(api):Created getSearch method to make search request and added thhis one within items request to get related items

nkrmr committed:
feat(api):Created getSearch method to make search request and added thhis one within items request to get related items

nkrmr committed:
feat(api):Created getSearch method to make search request and added thhis one within items request to get related items

nkrmr committed:
feat(api):Created getSearch method to make search request and added thhis one within items request to get related items

nkrmr committed:
Removed narrow to feedback block

nkrmr committed:
Removed narrow to feedback block

nkrmr committed:
Removed narrow to feedback block

nkrmr committed:
Updated test

nkrmr committed:
Updated test

nkrmr committed:
Updated test

nkrmr committed:
Updated test

nkrmr committed:
test(frontend):Updated test after rebase

nkrmr committed:
test(frontend):Updated test after rebase

nkrmr committed:
test(frontend):Updated test after rebase

nkrmr committed:
test(frontend):Updated test after rebase

nkrmr committed:
test(frontend):Updated test after rebase

Commit:
fix(frontend): search autofocus (#1385)

Commit:
feat(search-ccn): get more results from SIRENE (#1371)

* feat(search-ccn): get more results from SIRENE

 - use 50 instead of 10 results from API SIRENE
 - use siret2idcc api v2 to prevent multiple calls
 - fix UI bug when searching by IDCC

* lint

* remove trash

revolunet commented on packages/code-du-travail-api/src/server/routes/items/index.js:
```diff
@@ -31,8 +32,14 @@ router.get("/items/:source/:slug", async ctx => {
+
+  // Get current item title to find related items
+  const { title } = item._source;
+  console.log("Title is :", title);
```
can be removed

nkrmr replied:
Oups yes

revolunet commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -45,97 +47,72 @@ function Answer({
+                  .map(link => (
+                    <li key={link.title}>
+                      <Link
+                        href={`/${getRouteBySource(link.source)}/${link.slug}`}
```
i guess you need to add a `as` prop too ? 

like https://github.com/SocialGouv/code-du-travail-numerique/blob/master/packages/code-du-travail-frontend/src/search/SearchResults/Results.js#L36-L49

lionelB replied:
In this case, the href value should be `/${getRouteBySource(link.source)}/[slug]` 
the as value should be `/${getRouteBySource(link.source)}/${link.slug}`

nkrmr replied:
Done

revolunet commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -195,6 +220,65 @@ const StyledTooltip = styled(Tooltip)`
+    return function cleanEffect() {};
+  }, [children, html]);
+  return portalComponents;
+}
```
thanks for extracting; can this be exported to its own file ?

lionelB replied:
and rename useGlossary() ;)

nkrmr replied:
Done

renovate[bot] committed:
fix(deps): update dependency next to ^9.0.8 (#1391)

renovate[bot] committed:
chore(deps): update dependency rollup to ^1.23.1 (#1389)

UnbearableBear commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -159,14 +136,62 @@ const DefinitonTerm = ({ term, definition }) => {
+`;
+
+const StyledContent = styled.div`
+  width: ${props => (props.isFull ? "80%" : "70%")};
```
isFull ? Je l'aurais plutôt appellée "hasResults"

nkrmr replied:
Done

renovate[bot] committed:
fix(deps): update dependency @socialgouv/kali-data to v1.2.12 (#1392)

UnbearableBear commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -159,14 +136,62 @@ const DefinitonTerm = ({ term, definition }) => {
   text-align: center;
 `;
 
+const StyledWrapper = styled(Container)`
```
Je l'aurais appellé `StyledContainer`plutôt pour rappeler qu'il s'agit du component `Container` et pas du component `Wrapper`

nkrmr replied:
Le problème est qu'il y a déjà un composant StyledContainer

UnbearableBear replied:
Appelle l'autre StyledErrorContainer plutôt dans ce cas, ça fera l'affaire je pense

nkrmr replied:
OK

UnbearableBear commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -159,14 +136,62 @@ const DefinitonTerm = ({ term, definition }) => {
+  display: flex;
+  justify-content: space-around;
+  flex-wrap: wrap;
+  padding: 0;
```
tu dois vraiment enlever le padding sur ce Container ? Pas de soucis d'alignements si tu fais ça ?

nkrmr replied:
Oui car il est déjà dans un container

UnbearableBear commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -159,14 +136,62 @@ const DefinitonTerm = ({ term, definition }) => {
+  width: 30%;
+  color: ${colors.blue};
+  @media (max-width: ${breakpoints.tablet}) {
+    padding: 4rem 2rem 7rem 0;
```
Hum ça fiat beaucoup de nombres magiques tout ça ! Uilise les variabels d'espacement de l'ui plz 🙏

UnbearableBear commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -159,14 +136,62 @@ const DefinitonTerm = ({ term, definition }) => {
+`;
+
+const StyledMenu = styled.div`
+  padding: 2rem 0;
```
pareil ici 🙏

nkrmr replied:
Done

UnbearableBear commented:
Est-ce que tu pourrais rajouter des screenshots du rendu desktop / mobile pour avoir un quick overview de la feature ?

lionelB commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -45,90 +47,63 @@ function Answer({
-          });
-        });
-      });
+  const portalComponents = usePortals(children, html);
```
on devrait plutot appeller ça `const glossaryItems = useGlossary(children, html)`

nkrmr replied:
Done

lionelB commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -45,90 +47,63 @@ function Answer({
-      });
+  const portalComponents = usePortals(children, html);
+  const linkedResults = groupByDisplayCategory(relatedItems);
+  console.log("CONSOLE LOG: linkedResults", linkedResults);
```
à supprimer

lionelB commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -159,14 +134,50 @@ const DefinitonTerm = ({ term, definition }) => {
+`;
+
+const StyledMenuList = styled.div`
+  position: -webkit-sticky;
```
styled-component should handle this for us. it is not he  case for sticky ?

nkrmr replied:
Done

lionelB reviewed (commented):
I guess some file are missing here (I don't find groupByDisplayCategory function)

lionelB committed:
chore(release): add v3.3.0 changelog & version (#1387)

UnbearableBear commented:
> I guess some file are missing here (I don't find groupByDisplayCategory function)

It's already in the repo in `src/searc/utils`

lionelB commented:
> It's already in the repo in `src/searc/utils`

time to grab a coffee :)

Commit:
feat: add glossaire in footer (#1404)

* feat(footer): add glossaire link fix #1372

* tests: update snaps

Commit:
fix(search-ccn): better handle idcc search (#1400)

Commit:
fix(polyfill): add Object.values for Edge (#1393)

* fix(polyfill): add Object.values

https://sentry.tools.factory.social.gouv.fr/incubateur/code-du-travail-numerique/issues/1646/?query=is:unresolved

* fix(polyfill): remove fetch

* fix(themes): add isomorphic-unfetch

* fix(frontend): add missing isomorphic-unfetch

Commit:
fix: search convention cosmetics (#1406)

* fix(search-ccn): cosmetics for #1405

* tests(seach-ccn): snapshots

lionelB committed:
test(frontend): fix fetch mock (#1408)

lionelB commented on .gitignore:
```diff
@@ -7,4 +7,4 @@ __pycache__
 packages/code-du-travail-data/dump.json
 package-lock.json
 yarn-error.log
-
```
should be in your global .gitignore :)

nkrmr replied:
Done

lionelB commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -45,97 +47,72 @@ function Answer({
-        });
-      });
+  const portalComponents = usePortals(children, html);
+  const linkedResults = groupByDisplayCategory(relatedItems);
```
Since we are only interested in a subset of the results, why not filter the list befor on the api side ?

UnbearableBear replied:
I'm quite sure we will add related themes and so very soon in the UI too

lionelB replied:
In fact, we will probably use other methods than `getSearch()` to retrieve relatedTheme or relatedTools that's why I think it should be done on the API side

nkrmr replied:
It's better if it can be done on API Side!

nkrmr committed:
feat(frontend): Fixed PR issues

nkrmr committed:
feat(frontend): Fixed PR issues

nkrmr committed:
feat(frontend): Fixed PR issues

nkrmr committed:
test(frontend): Updated snapshots

nkrmr committed:
Merge branch 'nkrmr/related-items' of github.com:SocialGouv/code-du-travail-numerique into nkrmr/related-items

nkrmr committed:
test(frontend): Updated snapshots

Commit:
fix(CI): add ENVIRONMENT=production (#1409)

* fix(CI): add ENVIRONMENT=production

* Update deploy.yml

nkrmr committed:
test(frontend):Updated snapshots

nkrmr committed:
test(frontend):Updated snapshots

nkrmr committed:
Merge branch 'nkrmr/related-items' of github.com:SocialGouv/code-du-travail-numerique into nkrmr/related-items
